my-app/src/Map.js

```diff
@@ -348,15 +349,46 @@ export default function Map() {
       endingTimeArray
     );
 
+    const [busRouteName, startingStopName, endingStopName] = await getRouteInfo(
+      RouteID,
+      startingStop,
+      endingStop
+    );
+    
     drawNavRoute(
       startGeodata.features[0].center,
       endGeodata.features[0].center,
       startingStop,
       endingStop,
-      RouteID
+      busRouteName,
+      startingStopName,
+      endingStopName
     );
   }
 
+  async function getRouteInfo(RouteID, startingStop, endingStop){
+    const busRouteInfo = await fetch(
+      "https://catbus.ridesystems.net/Services/JSONPRelay.svc/GetRoutesForMap"
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        return data;
+      })
```

Comment:
This data has already been fetched and stored in `busRoutes`, which is component state. I would access this instead of fetching the data again.